### PR TITLE
Separate imagestreams and image listings

### PR DESCRIPTION
**Pull request by petervo**
Both types having the own listings. Implement the panels so these widgets can be used in cockpit.

https://raw.githubusercontent.com/cockpit-project/cockpit-design/0d577dd35fe946bc14a5d716c6d11394df5f79d4/kubernetes/image-registry-v3.png

### CODE REVIEW

**Comment by thomasmckay** [comment]
ACK tested w/ https://github.com/cockpit-project/cockpit/pull/7400

File: views/imagestream-panel.html [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        <div ng-if="actions" class="listing-ct-actions">
+            <button class="btn btn-danger btn-delete pficon pficon-delete"
+                ng-if="actions.deleteImageStream"
+                ng-click="actions.deleteImageStream(imagestream)"></button>
+            <button class="btn btn-default pficon pficon-edit"
+                ng-if="actions.modifyImageStream"
+                ng-click="actions.modifyImageStream(imagestream)"></button>
+        </div>
+        <ul class="nav nav-tabs nav-tabs-pf">
+            <li ng-class="{active: tab('main')}"><a ng-click="tab('main', $event)">Image stream</a></li>
```
Comment (stefwalter) [inline views/imagestream-panel.html]: Needs 'translate'

File: views/imagestream-panel.html [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+            <li ng-class="{active: tab('meta')}" ng-if="imagestream.metadata.annotations">
+                <a ng-click="tab('meta', $event)">Metadata</a></li>
```
Comment (stefwalter) [inline views/imagestream-panel.html]: Needs 'translate'

File: views/imagestream-panel.html [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+            <li ng-class="{active: tab('tags')}">
+                <a ng-click="tab('tags', $event)">Tags</a></li>
```
Comment (stefwalter) [inline views/imagestream-panel.html]: Needs 'translate'

File: views/imagestream-listing.html [hunk 4]
```diff
@@ -2,99 +2,45 @@
...
+                <a ng-repeat="tag in imagestreamTags(imagestream) | orderBy : 'tag.name' | limitTo: 4" class="registry-image-tag" ng-click="imagestreamActivate(imagestream, tag, $event)" title="{{tag.name }}">{{ tag.name }}</a>
+                <span ng-if="tag_count > 4" translate-n="tag_count - 4" translate translate-plural="and {{ $count }} others">and {{ $count }} other</span>
             </td>
             <td>
-                <div title="{{ tag.items[0].created }}">
-                    <span ng-if="tag.status.items.length &amp;&amp; tag.status.items[0].image" title="{{ tag.items[0].created }}">
-                        {{ tag.status.items[0].created | dateRelative }}
-                    </span>
-                </div>
+                  <span ng-if="!imageStream.status.dockerImageRepository && !imageStream.spec.dockerImageRepository"><em>unknown</em></span>
```
Comment (stefwalter) [inline views/imagestream-listing.html]: Needs translate

File: images.less [hunk 5]
```diff
@@ -30,10 +30,11 @@ a.registry-image-tag {
     text-decoration: none;
     max-width: 76px;
     height: 22px;
-    display: block;
+    display: inline;
     text-align: center;
-    padding: 0 2px;
+    padding: 0 5px;
     margin-left: 2px;
+    font-weight: normal;
```
Comment (stefwalter) [inline images.less:37]: The corners here need to be rounded properly.

**Review by stefwalter — APPROVED** [review]
Lots of code moved around, I hope I got the review right. This looks good to me with a few fixes. I'll push a fixup. Translate tags, in addition updated the CSS to match design.

**Comment by martinpitt** [comment]
The rounded corners match the [design](https://raw.githubusercontent.com/cockpit-project/cockpit-design/0d577dd35fe946bc14a5d716c6d11394df5f79d4/kubernetes/image-registry-v3.png) and the `translate` tag attributes look correct too, so ack on @stefwalter's fixup.

**Pull request closed without merging**